projects/OctoberSky1/config/UOStateMachine.cpp

```diff
@@ -294,16 +303,48 @@ void UOStateMachine::detectMotorBurnout(const sensorsData *data)
 		consecutiveEvents = 0;
 	}
 
-	// trigger appogee if the sbg detects "ApogeeThreshold" number of consecutive times
-	// that the rocket is pointing downwards and falling
-	if (consecutiveEvents >= ApogeeThreshold)
+	// trigger motor burn out if the sbg detects "BurnOutThreshold" number of consecutive times
+	// that the rocket is in coast
+	if (consecutiveEvents >= BurnOutThreshold)
 	{
 		std::cout << "MotorBurnout \n";
 		MotorBurnout();
 	}
 #endif
 }
 
+void UOStateMachine::detectTouchdown(const sensorsData *data)
+{
+#if USE_SBG
+	// TODO: only check for apogee x seconds after launch
+	// Euler angle
+	// pitch is pitch
+	static uint8_t consecutiveEvents = 0;
+
+	float xAcc = data->sbg.filteredXaccelerometer;
+	float yAcc = data->sbg.filteredYaccelerometer;
+	float zAcc = data->sbg.filteredZaccelerometer;
+
+	float resultingAcc = sqrt(pow(xAcc, 2) + pow(yAcc, 2) + pow(zAcc, 2));
+	if (resultingAcc <= -9.8)
```

Comment:
Because it is the resulting vector, it needs to be >= 9.8
Also, we could have a margin of 10%, so it would be (resultingAcc >= 8.82)